### PR TITLE
Handle NaNs and infs in KFoldCV scores

### PR DESCRIPTION
**Pull request by rcurtin**
This PR changes the behavior of `KFoldCV` to issue warnings when a given fold of k-fold cross-validation has a metric score that is either `nan` or `inf`, and then that score is ignored when averaging to return a result with the `Evaluate()` function.  If no scores were not `nan` or `inf`, then another warning is issued, and `Evaluate()` returns `0.0`.

This should solve #2565.

@micyril if you like, take a look, and let me know what you think of this solution or if I overlooked anything. :+1:

### CODE REVIEW

**Review by micyril — APPROVED** [review]
LGTM

**Review by zoq — APPROVED** [review]
Looks good to me as well.

File: HISTORY.md [hunk 1]
```diff
@@ -10,6 +13,7 @@
     methods/hoeffding_trees/information_gain.hpp (#2556).
 
   * Added macro for changing stream of printing and warnings/errors (#2556).
+>>>>>>> origin/master
```
Comment (zoq) [inline HISTORY.md:16]:
```suggestion
```

Comment (rcurtin) [reply HISTORY.md:16]: Thanks!  How did I miss that during the merge?  Oops.

**Comment by rcurtin** [comment]
Yeah, they are related to the R bindings.  If any of the builds fail here but it's clearly not related to the PR, I'll go ahead and merge anyway.  I think there may be a few problems with some of the other builds right now.

**Pull request merged**